### DISCUSSION TITLE
cascade styling for wbs nodes

### DISCUSSION THREAD

**Discussion by grivo** [plantuml/plantuml#1356] [Ideas]
as for now - we have two options for wbs styling
 - depths oriented
 - style name oriented

but, let's say, i want to apply style on some node and all it's children underneath?

i suppose, some flag sort of _applyOnChildren_ may be useful: it should work unless child node has it's own style declared

**Comment by The-Lum** [comment]
Hi @grivo, and all,

For that you can use `"descendant" "*" pseudo selector`:
```puml
@startwbs
<style>
wbsDiagram {
  .b * {
    BackgroundColor #00ccff
  }
}
</style>
* Root
** Node
*** Node
** Hello,  I am blue <<b>>
*** I am blue
**** and all my children
**** and all my children
*** I am blue
**** and all my children
** Node
*** Node
@endwbs
```
>[![](https://www.planttext.com/api/plantuml/svg/ZOyz2i9048NxFSK3kqKq1rX2K0lJM7Y3_KiCJdPWiq62vEwkGL9OYDLyyzxCOwOQaexfOGQXXZIHBqNcKwVRg7iy1R0pa0i01stlRUGnk2CJHsoAmjgcoTuiPg7sdm29AtCIKkB2pkUw0iwUYBT0ZPnkQFHGofJbChDgxqv21mTDX7w2lNNaeWy_ZV_slqwgV73vznS0)](https://www.planttext.com/?text=ZOyz2i9048NxFSK3kqKq1rX2K0lJM7Y3_KiCJdPWiq62vEwkGL9OYDLyyzxCOwOQaexfOGQXXZIHBqNcKwVRg7iy1R0pa0i01stlRUGnk2CJHsoAmjgcoTuiPg7sdm29AtCIKkB2pkUw0iwUYBT0ZPnkQFHGofJbChDgxqv21mTDX7w2lNNaeWy_ZV_slqwgV73vznS0)

- If that is what you expect?

See also doc. here:
- https://plantuml.com/fr/style-evolution#2334f7af067184e4
- https://forum.plantuml.net/14362/styling-descendents-error-in-wbs-diagrams

Regards.

**Comment by grivo** [comment]
oh, i accidentally missed it, thanks a lot!

**Reply by The-Lum** [reply]
_No problem._
Perhaps We can copy the examples of this page:
- https://plantuml.com/en/style-evolution#2334f7af067184e4

on this page:
- https://plantuml.com/en/wbs-diagram

_[and:_
- https://plantuml.com/en/mindmap-diagram
_]_

Regards.

**Comment by grivo** [comment]
sure, it'll help a lot
